extensions/agroal/deployment/src/test/java/io/quarkus/agroal/test/AgroalVaultKv2ITCase.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.agroal.test;
+
+import static io.quarkus.vault.VaultTestExtension.testDataSource;
+
+import java.sql.SQLException;
+
+import javax.inject.Inject;
+
+import org.jboss.logging.Logger;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.DisabledOnOs;
+import org.junit.jupiter.api.condition.OS;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.agroal.api.AgroalDataSource;
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vault.VaultTestExtension;
+
+@DisabledOnOs(OS.WINDOWS)
+@ExtendWith(VaultTestExtension.class)
+public class AgroalVaultKv2ITCase {
```

Comment:
What is `Kv`? It's not clear for us laymen :).

Reply:
Key Value store.
that is the vault secret engine that you use to store confidential information.
https://www.vaultproject.io/api/secret/kv/index.html
probably the easiest one to use if you just want to store a password.
it is a hierarchical structure (like a filesystem), where you can define rbac ("create", "read", "update", "delete", "list", "deny") on paths through policies. each node of the tree can store a map of string->string
policies are then referred to from vault roles (eg: approle, or k8s role) that you use to authenticate.

Reply:
@vsevel may be worth renaming to VaultKeyValueService ? (I've checked the docs, while `KV` is used most of the time, it also says `It can be a generic Key-Value store`). Not sure, just a suggestion

Reply:
indeed I have tried to respect vault's naming.
but no problem, I will do it if that makes it clearer.

Reply:
@sberyozkin I have renamed the class to VaultKeyValueSecretEngine => made the KV part explicit, and renamed the Service by the more accurate SecretEngine.
I also renamed the readSecrets() into read(), because I felt it was getting redeundant with the interface name.

Reply:
@vsevel A new name looks a bit complex (quite long)...but I guess it is not a big deal, thanks

Reply:
Hi @vsevel I'm starting to regret I suggested to rename :-), it was a bit cryptic all right `VaultKVService` but it was compact and the docs are clear what it is about - every user working with HashiCorp Vault will know what 'KV' is. I mean, we can keep the new name for sure, but please consider how the same expanded name approach would work for other store, lets pick 'AWS' and `ActiveDirectory` I can imagine users asking for the former at least :-). If you find the names will become too long then please revert the name update, but keep otherwise. Thanks.
Look forward to seeing you contribution in the master soon :-)